### ISSUE TITLE
Modal Accessibility

### ISSUE THREAD

**Issue by LKessell** [SWRPG-Capstone/swrpg-fe#50]
Refactor "Update Skills" modal with better accessibility
- [ ] Trap focus within the modal when it is open
- [ ] Modal can be closed with the ESC key
- [ ] Elements outside the modal should be hidden from screen readers while it is open
- [ ] Elements outside the modal should not be scrollable when it is open
- [ ] Add proper aria labels to modal
